lib/omnibus/packagers/pkgng.rb

```diff
@@ -0,0 +1,574 @@
+    # @example
+    #   prefix "/usr/local"
+    #
+    # @param [String] val
+    #   a prefix path
+    #
+    # @return [String]
+    #   the prefix path
+    #
+    def prefix(val = NULL)
+      if null?(val)
+        @prefix || "/"
+      else
+        unless val.is_a?(String)
+          raise InvalidValue.new(:prefix, "be a String")
+        end
+
+        @prefix = val
+      end
+    end
+    expose :prefix
+
+    #
+    # Set or return the licenses for this package.
+    #
+    # @example
+    #   licenses ["Apache 2.0", "MIT"]
+    #
+    # @param [Array] val
+    #   the licenses for this package
+    #
+    # @return [Array]
+    #   the licenses for this package
+    #
+    def licenses(val = NULL)
+      if null?(val)
+        @licenses || [project.license]
+      else
+        unless val.is_a?(Array)
+          raise InvalidValue.new(:licenses, "be an Array")
+        end
+
+        @licenses = val
+      end
+    end
+    expose :licenses
+
+    #
+    # Set or return the categories for this package.
+    #
+    # @example
+    #   categories ["databases"]
+    #
+    # @param [Array] val
+    #   the categories for this package
+    #
+    # @return [Array]
+    #   the categories for this package
+    #
+    def categories(val = NULL)
+      if null?(val)
+        @categories || ["misc"]
+      else
+        unless val.is_a?(Array)
+          raise InvalidValue.new(:categories, "be an Array")
+        end
+
+        @categories = val
+      end
+    end
+    expose :categories
+
+    #
+    # Set or return the users for this package.
+    #
+    # @example
+    #   users ["root"]
+    #
+    # @param [Array] val
+    #   the users for this package
+    #
+    # @return [Array]
+    #   the users for this package
+    #
+    def users(val = NULL)
+      if null?(val)
+        @users || [project.package_user]
+      else
+        unless val.is_a?(Array)
+          raise InvalidValue.new(:users, "be an Array")
+        end
+
+        @users = val
+      end
+    end
+    expose :users
+
+    #
+    # Set or return the groups for this package.
+    #
+    # @example
+    #   groups ["root"]
+    #
+    # @param [Array] val
+    #   the groups for this package
+    #
+    # @return [Array]
+    #   the groups for this package
+    #
+    def groups(val = NULL)
+      if null?(val)
+        @groups || [project.package_group]
+      else
+        unless val.is_a?(Array)
+          raise InvalidValue.new(:groups, "be an Array")
+        end
+
+        @groups = val
+      end
+    end
+    expose :groups
+
+    #
+    # Set or return the dependencies for this package.
+    #
+    # @example
+    #   dependencies {
+    #     "libiconv" => {
+    #       "origin" => "converters/libiconv",
+    #       "version" => "1.13.1_2"
+    #     }
+    #   }
+    #
+    # @param [Hash] val
+    #   the dependencies for this package
+    #
+    # @return [Hash]
+    #   the dependencies for this package
+    #
+    def dependencies(val = NULL)
```

Comment:
If this is for runtime dependencies, I'd rather call it `runtime_dependencies`; the `omnibus` DSL uses `dependency` and `dependencies` for buildtime dependencies.